web/server/entrypoint.go

```diff
@@ -416,70 +506,27 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 	defer func() {
 		err = multierr.Combine(err, watcher.Close())
 	}()
-	onWatchDone := make(chan struct{})
-	oldCfg := processedConfig
-	utils.ManagedGo(func() {
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			default:
-			}
-			select {
-			case <-ctx.Done():
-				return
-			case cfg := <-watcher.Config():
-				processedConfig, err := processConfig(cfg)
-				if err != nil {
-					s.logger.Errorw("reconfiguration aborted: error processing config", "error", err)
-					continue
-				}
-
-				// flag to restart web service if necessary
-				diff, err := config.DiffConfigs(*oldCfg, *processedConfig, s.args.RevealSensitiveConfigDiffs)
-				if err != nil {
-					s.logger.Errorw("reconfiguration aborted: error diffing config", "error", err)
-					continue
-				}
-				var options weboptions.Options
-
-				if !diff.NetworkEqual {
-					// TODO(RSDK-2694): use internal web service reconfiguration instead
-					myRobot.StopWeb()
-					options, err = s.createWebOptions(processedConfig)
-					if err != nil {
-						s.logger.Errorw("reconfiguration aborted: error creating weboptions", "error", err)
-						continue
-					}
-				}
-
-				// Update logger registry if log patterns may have changed.
-				//
-				// This functionality is tested in `TestLogPropagation` in `local_robot_test.go`.
-				if !diff.LogEqual {
-					s.logger.Debug("Detected potential changes to log patterns; updating logger levels")
-					config.UpdateLoggerRegistryFromConfig(s.registry, processedConfig, s.logger)
-				}
-
-				myRobot.Reconfigure(ctx, processedConfig)
 
-				if !diff.NetworkEqual {
-					if err := myRobot.StartWeb(ctx, options); err != nil {
-						s.logger.Errorw("reconfiguration failed: error starting web service while reconfiguring", "error", err)
-					}
-				}
-				oldCfg = processedConfig
-			}
-		}
-	}, func() {
-		close(onWatchDone)
-	})
+	onWatchDone := make(chan struct{})
+	go func() {
+		defer func() {
+			close(onWatchDone)
```

Comment:
If we're leaving this as a raw goroutine -- can go straight to `defer close(onWatchDone)`.

Reply:
Yep.